core/src/main/java/org/owasp/dependencycheck/analyzer/PinnedMavenInstallAnalyzer.java

```diff
@@ -318,14 +352,106 @@ public String getCoord() {
+    private static final ObjectReader INSTALL_FILE_V2_READER;
+    private static final ObjectMapper MAPPER;
 
     static {
-        final ObjectMapper mapper = new ObjectMapper();
-        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-        INSTALL_FILE_READER = mapper.readerFor(InstallFile.class);
+        MAPPER = new ObjectMapper();
+        MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        INSTALL_FILE_READER = MAPPER.readerFor(InstallFile.class);
+        INSTALL_FILE_V2_READER = MAPPER.readerFor(InstallFileV2.class);
+    }
+
+    /**
+     * Represents the entire pinned Maven dependency set in an install.json
+     * file.
+     *
+     * <p>
+     * At the time of writing, the latest version is 2, and the dependencies
+     * are stored in {@code .artifacts}.
+     *
+     * <p>
+     * The top-level keys we care about are {@code .artifacts}. {@code .version}.
+     */
+    private static class InstallFileV2 {
+
+        /**
+         * The file format version.
+         */
+        @JsonProperty("version")
+        private String version;
+
+        /**
+         * A list of Maven dependencies made available. Note that this map is transitively closed and
+         * pinned to a specific version of each artifact.
+         * <p>
+         * The key is the Maven coordinate string, less the version
+         * {@code group:artifact[:optional classifier][:optional packaging]}.
+         * <p>
+         * The value contains the version of the artifact.
+         */
+        @JsonProperty("artifacts")
+        private Map<String, Artifactv2> artifacts;
+
+        /**
+         * A sentinel value placed in the file to indicate that it is an auto-generated pinned maven
+         * install file.
+         */
+        @JsonProperty("__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY")
+        private String autogeneratedSentinel;
+
+        /**
+         * Returns artifacts.
+         *
+         * @return artifacts
+         */
+        public Map<String, Artifactv2> getArtifacts() {
+            return artifacts;
+        }
+
+        /**
+         * Returns version.
+         *
+         * @return version
+         */
+        public String getVersion() {
+            return version;
+        }
+
+        /**
+         * Returns autogeneratedSentinel.
+         *
+         * @return autogeneratedSentinel
+         */
+        public String getAutogeneratedSentinel() {
+            return autogeneratedSentinel;
+        }
+    }
+    private static class Artifactv2 {
+
+          /**
+          * The version of the artifact.
+          */
+          @JsonProperty("version")
+          private String version;
+
+          /**
+          * Returns the value of version.
+          *
+          * @return the value of version
+          */
+          public String getVersion() {
+              return version;
+          }
```

Comment:
You could implement `HasCoord` here instead of doing so anonymously up above. Maybe a little more obvious.